tests/test_impls.py

```diff
@@ -43,3 +47,25 @@ def check_initial_contents(filename):
 
         assert not os.path.exists(bkp_commit_file)
         check_initial_contents(commit_file)
+
+
+@pytest.mark.parametrize(
+    ["initial_req", "cuda_suffix", "cuda_major", "expected_req"],
+    [
+        ("cupy", "", "11", "cupy-cuda11x"),
+        ("cupy", "", "12", "cupy-cuda12x"),
+        ("cupy", "", None, "cupy"),
+        ("cuda-python", "", "11", "cuda-python<12.0.0.dev0,>=11.0.0"),
+        ("cuda-python>=11.7.1", "", "12", "cuda-python<13.0.0.dev0,>=11.7.1,>=12.0.0"),
```

Comment:
Can we canonicalize this somehow? We don't need both `>=11.7.1` and `>=12.0.0`.

Also I don't like that the `<` is placed before the `>=`. That's not how versions are usually written.

Reply:
Rendered irrelevant, since we're no longer checking `cuda-python`. This PR is now for tests only.